src/super_gradients/training/transforms/keypoints/keypoints_compose.py

```diff
@@ -0,0 +1,132 @@
+        """
+        if KeypointsMixup in self.transforms or KeypointsMosaic in self.transforms:
+            raise RuntimeError("KeypointsMixup & KeypointsMosaic augmentations are not supported in old-style transforms API")
+
+        for t in self.transforms:
+            image, mask, joints, areas, bboxes = t(image, mask, joints, areas, bboxes)
+
+        return image, mask, joints, areas, bboxes
+
+    def apply_to_sample(self, sample: PoseEstimationSample) -> PoseEstimationSample:
+        """
+        Applies the series of transformations to the input sample.
+        The function may modify the input sample inplace, so input sample should not be used after the call.
+
+        :param sample: Input sample
+        :return:       Transformed sample.
+        """
+        sample = sample.sanitize_sample()
+        sample = self._apply_transforms(
+            sample, transforms=self.transforms, load_sample_fn=self.load_sample_fn, min_bbox_area=self.min_bbox_area, min_visible_joints=self.min_visible_joints
+        )
+        return sample
+
+    @classmethod
+    def _apply_transforms(
+        cls, sample: PoseEstimationSample, transforms: List[AbstractKeypointTransform], load_sample_fn, min_bbox_area, min_visible_joints
+    ) -> PoseEstimationSample:
+        """
+        This helper method allows us to query additional samples for mixup & mosaic augmentations
+        that would be also passed through augmentation pipeline. Example:
+
+        ```
+          transforms:
+            - KeypointsBrightnessContrast:
+                brightness_range: [ 0.8, 1.2 ]
+                contrast_range: [ 0.8, 1.2 ]
+                prob: 0.5
+            - KeypointsHSV:
+                hgain: 20
+                sgain: 20
+                vgain: 20
+                prob: 0.5
+            - KeypointsLongestMaxSize:
+                max_height: ${dataset_params.image_size}
+                max_width: ${dataset_params.image_size}
+            - KeypointsMixup:
+                prob: ${dataset_params.mixup_prob}
+        ```
+
+        In the example above all samples in mixup will be forwarded through KeypointsBrightnessContrast, KeypointsHSV,
+        KeypointsLongestMaxSize and only then mixed up.
+
+        :param sample:         Input data sample
+        :param transforms:     List of transformations to apply
+        :param load_sample_fn: A method to load additional samples if needed
+        :param min_bbox_area:  Min bbox area of the pose instances to keep them
+        :return:               Transformed sample
+        """
+        applied_transforms_so_far = []
+        for t in transforms:
+            if t.additional_samples_count == 0:
+                sample = t.apply_to_sample(sample)
+                applied_transforms_so_far.append(t)
+            else:
+                additional_samples = [load_sample_fn() for _ in range(t.additional_samples_count)]
```

Comment:
From what I understand load_sample_fn should point to the dataset's load_sample method, correct?
In that case load_sample expects an independent as an argument (from what I saw at super_gradients.training.datasets.pose_estimation_datasets.base_keypoints.BaseKeypointsDataset.load_sample).
If I am right, does it work like this?
Otherwise maybe lets generate an index?

Reply:
Initially I did it via `load_sample_fn(index)` but then I though this is not great since we have to pass a dataset size to KeypointsCompose to figure out the range of indexes we can sample from.
Ended up with a argument-free method instead because decided this is too much knowledge exposed to Compose. WDYT?

Reply:
So the existing dataset classes for pose estimation will not support the loading of additional samples (This is also helps gently push users to migrate to new sample-centric datasets).
And for new dataset what I did is this:
```
    @abc.abstractmethod
    def load_sample(self, index: int) -> PoseEstimationSample:
        """
        Read a sample from the disk and return a PoseEstimationSample
        :param index: Sample index
        :return: Returns an instance of PoseEstimationSample that holds complete sample (image and annotations)
        """
        raise NotImplementedError()

    def load_random_sample(self):
        num_samples = len(self)
        random_index = random.randrange(0, num_samples)
        return self.load_sample(random_index)

```
To me it's more clear to have two methods other than one with optional parameter, since in latter case we are asking user to not only implement dataset-specific logic of loading data but also a sampling logic which IMO should be common for all dataset classes.

Reply:
Marking as resolved after discussing with @shaydeci privately ^_^

Reply:
How will we pass load_random_sample in Yaml config though ?

Reply:
Any answer on this? How did you handle this in the yaml file when running ?
It's my last open comment...

Reply:
Pardon, completely missed your question ^_^
We don't need to pass KeypointsCompose explicitly in YAML. In fact, KeypointsCompose not even in the registry so it is technically impossible to instantiate it in a YAML. 
It is created internally inside base dataset class and thus able to obtain a reference to `load_random_sample`.